### PR TITLE
Added federation ping, loopback and dashboard

### PR DESCRIPTION
**Pull request by alansikora**

### CODE REVIEW

File: packages/rocketchat-i18n/i18n/en.i18n.json [hunk 1]
```diff
@@ -2751,7 +2753,7 @@
   "The_message_is_a_discussion_you_will_not_be_able_to_recover": "The message is a discussion you will not be able to recover the messages!",
   "The_peer__peer__does_not_exist": "The peer <em>__peer__</em> does not exist.",
   "The_redirectUri_is_required": "The redirectUri is required",
-  "The_server_will_restart_in_s_seconds": "The server will restart in %s seconds",
+  "The server will restart in %s seconds": "The server will restart in %s seconds",
```
Comment (engelgabriel) [inline packages/rocketchat-i18n/i18n/en.i18n.json]: I think you have damaged the translation key here

Comment (alansikora) [reply packages/rocketchat-i18n/i18n/en.i18n.json]: Oh, damn. Sorry about that, fixed.

File: server/startup/migrations/v142.js [hunk 2]
```diff
@@ -0,0 +1,20 @@
+import { Migrations } from '../../../app/migrations/server';
+
+import { Users, FederationPeers } from '../../../app/models/server';
+
+Migrations.add({
+	version: 142,
+	up() {
+		const users = Users.find({ federation: { $exists: true } }).fetch();
```
Comment (rodrigok) [inline server/startup/migrations/v142.js]: Can you select only the `federation` fields?

File: app/federation/server/PeerHTTP/PeerHTTP.js [hunk 3]
```diff
@@ -62,33 +67,43 @@ export class PeerHTTP {
 
 						retryInfo.DNSUpdated = true;
 
-						this.log(`Trying to update local DNS cache for peer:${ peer.domain }`);
+						this.log(
```
Comment (rodrigok) [inline app/federation/server/PeerHTTP/PeerHTTP.js]: Why breaking line on this things?

File: app/models/server/models/FederationPeers.js [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+class FederationPeersModel extends Base {
+	constructor() {
+		super('federation_peers');
+	}
+
+	refreshPeers() {
+		const collectionObj = this.model.rawCollection();
+		const findAndModify = Meteor.wrapAsync(collectionObj.findAndModify, collectionObj);
+
+		const users = Users.find({ federation: { $exists: true } }).fetch();
```
Comment (rodrigok) [inline app/models/server/models/FederationPeers.js]: Should return only the federation field

File: app/federation/server/PeerClient.js [hunk 5]
```diff
@@ -329,7 +329,9 @@ export class PeerClient {
 		const { peer: { domain: localPeerDomain } } = this;
 
 		// Check if room is federated
-		if (!FederatedRoom.isFederated(localPeerDomain, room, { checkUsingUsers: true })) { return roomOwner; }
+		if (!FederatedRoom.isFederated(localPeerDomain, room, { checkUsingUsers: true })) {
```
Comment (rodrigok) [inline app/federation/server/PeerClient.js]: Revert

File: app/federation/server/PeerDNS.js [hunk 6]
```diff
@@ -138,7 +142,13 @@ export class PeerDNS {
 				throw new Error(`Error trying to fetch SRV DNS entries for ${ domain }`);
 			}
 
-			peer = this.getPeerUsingHub(domain);
+			try {
+				peer = this.getPeerUsingHub(domain);
+			} catch (err) {
+				throw new Error(
```
Comment (rodrigok) [inline app/federation/server/PeerDNS.js]: Remove line breaks

File: app/federation/server/index.js [hunk 7]
```diff
@@ -61,10 +67,15 @@ peerClient.start();
...
 peerServer.start();
 
+// Start the pinger, to check the status of all peers
+peerPinger.start();
+
 const updateSettings = _.debounce(Meteor.bindEnvironment(function() {
 	const _enabled = settings.get('FEDERATION_Enabled');
 
-	if (!_enabled) { return; }
+	if (!_enabled) {
```
Comment (rodrigok) [inline app/federation/server/index.js]: Revert

**Pull request merged**